src/controllers/discussionController.ts

```diff
@@ -0,0 +1,266 @@
+import { Request, Response, NextFunction } from "express";
+import Discussion, { IDiscussion } from "../models/Discussion";
+import { StatusCodes } from "http-status-codes";
+import {
+  BadRequestError,
+  NotFoundError,
+  UnauthenticatedError,
+} from "../errors";
+import Joi from "joi";
+import mongoose from "mongoose";
+import { IUser } from "../models/User";
+
+const discussionJoiSchema = Joi.object({
+  title: Joi.string().required(),
+  book: Joi.string().required(),
+  content: Joi.string().required(),
+  date: Joi.date().iso().required(),
+  participants: Joi.array().items(Joi.string()).optional(),
+  meetingLink: Joi.string().uri().required(),
+  createdBy: Joi.string().required(),
+});
+
+interface IGetDiscussionsQuery {
```

Comment:
you can create an interface file to store all your interfaces

Reply:
done